src/core/weightedGraph.js

```diff
@@ -76,3 +77,108 @@ export function overrideWeights(
   });
   return {graph: wg.graph, weights};
 }
+
+export type NodeWithWeight = {|
+  +node: Node,
+  +weight: ?Weights.NodeWeight,
```

Comment:
There's two concepts of weight for a node or edge in the graph. The "own weight" which is the weight directly assigned to this object (which may be undefined), and the "evaluated weight" which is the implied weight from the structure of the weighted graph (which is always defined, and defaults to 1 in the case of nodes). Should take care to define which one we're talking about here.